### PR TITLE
fix(swarm)!: make on_connection_handler_event impl mandatory.

### PR DESCRIPTION
**Pull request by jxs**
## Description
Sorry I missed this on https://github.com/libp2p/rust-libp2p/pull/3264

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]
Can you expand the changelog entry of #3264 to say that this is now mandatory?

**Comment by jxs** [comment]
> Can you expand the changelog entry of #3264 to say that this is now mandatory?

Yeah makes sense, updated!

File: swarm/CHANGELOG.md [hunk 1]
```diff
@@ -9,19 +9,21 @@
 - Add `estblished_in` to `SwarmEvent::ConnectionEstablished`. See [PR 3134].
 
 - Remove deprecated `inject_*` methods from `NetworkBehaviour` and `ConnectionHandler`.
-  see [PR 3264].
+  Make the implementation of `on_swarm_event` and `on_connection_handler_event`
+  both mandatory. See [PR 3264] and [PR 3364].
```
Comment (mxinden) [inline swarm/CHANGELOG.md:13]:
```suggestion
  Make the implementation of `on_connection_handler_event`
  mandatory and thus consistent with `on_swarm_event`. See [PR 3264] and [PR 3364].
```

I find the wording hard to understand, i.e. I would expect this change to touch both methods, whereas in reality it doesn't.

**Review by mxinden — APPROVED** [review]
One wording suggestion. Otherwise looks good to me.

File: swarm/CHANGELOG.md (hunk 1 above)
Comment (jxs) [reply swarm/CHANGELOG.md:13]: sorry for the confusion introduced Max. It does, it's just that for `on_connection_handler_event` it was left out on #3264. Does the CHANGELOG entry make more sense to you, or do you still want me to adress it?

Comment (mxinden) [reply swarm/CHANGELOG.md:13]: Ah, right, my bad. Thanks for clarifying.

**Review by mxinden — APPROVED** [review]

**Pull request merged**